pageserver/src/page_service.rs

```diff
@@ -849,67 +849,47 @@ impl PageServerHandler {
     /// the LSN that should be used to look up the page versions.
     async fn wait_or_get_last_lsn(
```

Comment:
I added this comment. Sorry if it is not quite clear but I did my nest to explain why do we need range.

Reply:
I find these 'horizon' and 'lsn' to be an unnatural way of thinking about this. My mental model is:

- the Compute always wants to see the data at one particular LSN. For a standby, it's the last-replayed LSN. For the primary, it's the last inserted LSN . Let's call this the "request LSN".
- In addition to the request LSN, the compute may send an additional hint to the pageserver. The hint tells the pageserver, "I know that the page hasn't been modified between the hint LSN, and the request LSN". If the compute doesn't know of any better, it can just use the request LSN for both.

I propose we call the fields in the Request structs, and elsewhere:
- simply `lsn` for the request LSN.
- `not_modified_since` for that hint.

Reply:
> My mental model is:

Yeah, that's much clearer.